content/en/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -90,57 +119,133 @@ You can develop a KMS plugin gRPC server using a stub file available for Go. For
-  to develop the gRPC server code 
+
+* Using Go: A high level
+  [library](https://github.com/kubernetes/kms/blob/release-{{< skew currentVersion >}}/pkg/service/interface.go)
+  is provided to make the process easier.  Low level implementations
+  can use the functions and data structures in the stub file:
+  [api.pb.go](https://github.com/kubernetes/kms/blob/release-{{< skew currentVersion >}}/apis/v2/api.pb.go)
+  to develop the gRPC server code
 
 * Using languages other than Go: Use the protoc compiler with the proto file:
-  [api.proto](https://github.com/kubernetes/kubernetes/blob/release-1.25/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v2alpha1/api.proto)
+  [api.proto](https://github.com/kubernetes/kms/blob/release-{{< skew currentVersion >}}/apis/v2/api.proto)
   to generate a stub file for the specific language
 
 Then use the functions and data structures in the stub file to develop the server code.
 
 #### Notes
 
 ##### KMS v1 {#developing-a-kms-plugin-gRPC-server-notes-kms-v1}
+
 * kms plugin version: `v1beta1`
 
   In response to procedure call Version, a compatible KMS plugin should return `v1beta1` as `VersionResponse.version`.
 
 * message version: `v1beta1`
 
-  All messages from KMS provider have the version field set to current version v1beta1.
+  All messages from KMS provider have the version field set to `v1beta1`.
 
 * protocol: UNIX domain socket (`unix`)
 
   The plugin is implemented as a gRPC server that listens at UNIX domain socket. The plugin deployment should create a file on the file system to run the gRPC unix domain socket connection. The API server (gRPC client) is configured with the KMS provider (gRPC server) unix domain socket endpoint in order to communicate with it. An abstract Linux socket may be used by starting the endpoint with `/@`, i.e. `unix:///@foo`. Care must be taken when using this type of socket as they do not have concept of ACL (unlike traditional file based sockets). However, they are subject to Linux networking namespace, so will only be accessible to containers within the same pod unless host networking is used.
 
 ##### KMS v2 {#developing-a-kms-plugin-gRPC-server-notes-kms-v2}
-* kms plugin version: `v2alpha1`
 
-  In response to procedure call Status, a compatible KMS plugin should return `v2alpha1` as `StatusResponse.Version`, "ok" as `StatusResponse.Healthz` and a keyID (KMS KEK ID) as `StatusResponse.KeyID`
+* KMS plugin version: `v2beta1`
+
+  In response to procedure call `Status`, a compatible KMS plugin should return `v2beta1` as `StatusResponse.version`,
+  "ok" as `StatusResponse.healthz` and a `key_id` (remote KMS KEK ID) as `StatusResponse.key_id`.
+
+  The API server polls the `Status` procedure call approximately every minute when everything is healthy,
+  and more frequently when the plugin is not healthy.  Plugins must take care to optimize this call as it will be
```

Comment:
may be beneficial to note the default value here.

Reply:
We currently mention the default value for status poll is every minute.

Reply:
> and more frequently when the plugin is not healthy
I meant to ask about the frequency when the plugin is not healthy.

Reply:
will add this